.github/workflows/ci.yml

```diff
@@ -1,8 +1,28 @@
-name: GitHub Actions Demo
-run-name: ${{ github.actor }} is testing out GitHub Actions 🚀
-on: [push]
+name: Build and Deploy Preview
+run-name: Deploying Preview for ${{ github.head_ref }}
+on:
```

Comment:
This action tests, builds, and deploys a preview for this PR.
This only triggers when the branch is opened or synchronized (pushed when the PR is open).